src/Core/UbuntuDriversBackend.vala

```diff
@@ -130,14 +125,28 @@ public class AppCenterCore.UbuntuDriversBackend : Backend, Object {
     }
 
     public async bool refresh_cache (Cancellable? cancellable) throws GLib.Error {
+        working = true;
+        string? command_output;
+        var result = yield get_drivers_output (cancellable, out command_output);
+        if (!result || command_output == null || cancellable.is_cancelled ()) {
+            working = false;
+            return false;
+        }
+
+        string[] tokens = command_output.split ("\n");
```

Comment:
You should filter out empty tokens from here, so that no empty strings get saved to settings. These empty strings are right now only checked when `get_installed_applications` is called.